### ISSUE TITLE
Block submissions in question preview for student-created variant

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#5418]
As discussed on Slack. From the issues page, the "student view" link leads to the student's version of the question page, which does not allow me to submit new answers as an instructor, which makes sense. However, the "instructor view" link leads to the question preview page and gets linked to the variant_id, which would allow the instructor using that page to submit new answers to the variant. This in essence has the side effect of linking any submission to the student's submission. I believe this is undesirable behaviour.

Alternatives:
* Changing the "instructor view" link in the issues page to use variant_seed instead of variant_id (this is already done for staff with no student data access anyway). This means that the instructor would see, in instructor view, the same params being generated for the question (assuming nothing changed in generate), but not the student's answer, issues and other related data. @mwest1066 points out that this would not be ideal since although the seed is the same, this would not allow for easy debugging of broken variants, since changes in generate would affect the params used for the question.
* Continue to use variant_id, but add the same restriction regarding submissions as there are in the student's version of the instance question. This has the benefit that all information associated to the student's view is the same (submission history, issues, etc.), but also limits debugging since it does not allow additional submissions to be used for testing.
* Provide a way to clone the variant by copying all question data from generate (params, true_answer, etc.) instead of the same seed. To be decided: should all submissions also be copied? (I don't think so).

I'd personally prefer to see both 2 (to ensure remaining links to that variant don't remain and cause trouble) and 3 implemented.

**Comment by mwest1066** [comment]
I agree that both (2) and (3) seem like the right thing to do here. That is, we initially show (2) but with submissions blocked and a button that says "clone these parameters" to implement (3).

**Comment by jonatanschroeder** [comment]
Option 3 might also be useful in the instructor view links from the assessment instance page, as well as the QID link from the instructor panel of student view.

**Comment by jonatanschroeder** [comment]
> > Option 3 might also be useful in the instructor view links from the assessment instance page, as well as the QID link from the instructor panel of student view.
> 
> What do those links do now? I'm a bit hesitant to make too many magic links that copy params without saying what they are doing, but I very much support giving instructors clear ways to view the different variant options (a student's exact variant, a cloned variant with the same data, a new variant with the same seed, a new variant with a different seed) and to understand what kind of variant they are selecting.

The instructor view link in the assessment instance page links to the question, but without specifying a variant or seed (i.e., the question at a random variant).

The QID link in the instructor panel links to the question preview with the same variant seed as the student's variant.

**Comment by mwest1066** [comment]
> Proposal: two buttons ("New variant" and "Clone variant") in the question footer for an instance question viewed as staff:

I like this!
